src/Features/Core/Portable/PullMemberUp/QuickAction/AbstractMemberPullerWithQuickAction.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.Editing;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+
+namespace Microsoft.CodeAnalysis.PullMemberUp.QuickAction
+{
+    internal abstract class AbstractMemberPullerWithQuickAction
+    {
+        /// <summary>
+        ///  This method is used to check whether the selected member overrides the member in destination.
+        ///  It just checks the members directly declared in the destination.
+        /// </summary>
+        protected abstract bool IsSelectedMemberDeclarationAlreadyInDestination(INamedTypeSymbol destination, ISymbol symbol);
+
+        internal CodeAction TryComputeRefactoring(
+            Document document,
+            ISymbol selectedMember,
+            INamedTypeSymbol destinationType,
+            CancellationToken cancellationToken)
+        {
+            var result = PullMembersUpAnalysisBuilder.BuildAnalysisResult(destinationType, ImmutableArray.Create(selectedMember));
+            if (result.PullUpOperationCausesError ||
+                IsSelectedMemberDeclarationAlreadyInDestination(destinationType, selectedMember))
+            {
+                return default;
+            }
+
+            return TryGetCodeAction(result, document, cancellationToken);
+        }
+
+        internal CodeAction TryGetCodeAction(
+            PullMembersUpAnalysisResult result,
+            Document contextDocument,
+            CancellationToken cancellationToken)
+        { 
+            if (result.Destination.TypeKind == TypeKind.Interface)
+            {
+                return new DocumentChangeAction(
+                    string.Format(FeaturesResources.Add_to_0, result.Destination),
+                    token => PullMembersIntoInterfaceAsync(result, contextDocument, token));
+            }
+            else if (result.Destination.TypeKind == TypeKind.Class)
+            {
+                return new SolutionChangeAction(
+                    string.Format(FeaturesResources.Add_to_0, result.Destination),
+                    token => PullMembersUpAsync(result, contextDocument, token));
```

Comment:
it's somewhat surprising that pulling into an interface is only a Doc change, but pulling into a Class is a Solution change.

Reply:
It is a document change since it just moves the declaration into one document and don't edit anything else for quick action. This part is just the simplify version from the dialog PR, since that scenario is more complex than this one. The original syntax may need change so it also return a changedSolution.

COuld we just keep this and after the we finish dialog PR they will all return SolutionChangedAction

Reply:
> COuld we just keep this and after the we finish dialog PR they will all return SolutionChangedAction

sure.  i wasn't suggesting this needed ot be changed :)

Note: to clarify.  A SolutionChange and a DocumentChange are effectively the same thing.  That's because a Solution *is* a snapshot all the way down to the Document level (and vice-versa).  So this was more about consistency.  If one was going to be a SolutionChangeAction, i'd just have both of them be that way.  But it's a ***really*** minor point.  I wasn't trying to suggest you needed to actually do this :)